### PR TITLE
fix(react-grid): correctly calculate rows to load when sorting/filtering in lazy loading mode 

### PR DESCRIPTION
**Pull request by LazyLahtak**
Fixes #[2114](https://github.com/DevExpress/devextreme-reactive/issues/2114)

### CODE REVIEW

File: packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx [hunk 1]
```diff
@@ -34,15 +34,18 @@ class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps,
...
+    { virtualRows }: Getters,
   ) => {
     const { pageSize, totalRowCount } = this.props;
 
     let newBounds;
     let requestedRange;
     let actualVirtualRows = virtualRows;
     if (forceReload) {
-      newBounds = requestedRange = { start: skip, end: skip + pageSize! * 2 };
+      const loadedInterval = intervalUtil.getRowsInterval(virtualRows);
```
Comment (ushkal) [inline packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx]: loadedInterval is declared twice

File: packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx [hunk 2]
```diff
@@ -34,18 +34,22 @@ class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps,
...
     const { pageSize, totalRowCount } = this.props;
 
     let newBounds;
     let requestedRange;
     let actualVirtualRows = virtualRows;
+    let loadedInterval;
     if (forceReload) {
-      newBounds = requestedRange = { start: skip, end: skip + pageSize! * 2 };
+      loadedInterval = intervalUtil.getRowsInterval(virtualRows);
+      newBounds = requestedRange = getForceReloadInterval(
```
Comment (MaximKudriavtsev) [inline packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx]: Why we are using this notation `a = b = value` and are not using `let a, b, c;`?

I think these things are the same. We should not use `a = b = value`

File: packages/dx-grid-core/src/plugins/virtual-table-state/helpers.test.ts [hunk 3]
```diff
@@ -326,4 +326,20 @@ describe('VirtualTableState helpers', () => {
       });
     });
   });
+
+  describe('#getForceReloadInterval', () => {
+    it('should return loadedInterval.start + pageSize * 2 as end', () => {
```
Comment (ushkal) [inline packages/dx-grid-core/src/plugins/virtual-table-state/helpers.test.ts]: ... if loaded interval is less than 2 pages

Comment (ushkal) [reply packages/dx-grid-core/src/plugins/virtual-table-state/helpers.test.ts]: should return 2 pages if loaded interval is less than 2 pages

File: packages/dx-grid-core/src/plugins/virtual-table-state/helpers.test.ts [hunk 4]
```diff
@@ -326,4 +326,20 @@ describe('VirtualTableState helpers', () => {
...
+      expect(getForceReloadInterval({ start: 100, end: 200 }, 100, 1000)).toEqual({
+        start: 100,
+        end: 300,
+      });
+    });
+
+    it('should return loadedInterval.end as end', () => {
```
Comment (ushkal) [inline packages/dx-grid-core/src/plugins/virtual-table-state/helpers.test.ts]: should return loaded interval if it is more than 2 pages

File: packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx [hunk 5]
```diff
@@ -34,15 +34,18 @@ class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps,
...
   ) => {
     const { pageSize, totalRowCount } = this.props;
 
     let newBounds;
     let requestedRange;
     let actualVirtualRows = virtualRows;
     if (forceReload) {
-      newBounds = requestedRange = { start: skip, end: skip + pageSize! * 2 };
+      const loadedInterval = intervalUtil.getRowsInterval(virtualRows);
+      newBounds = requestedRange = getDoublePageInterval(
```
Comment (ushkal) [inline packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx]: loaded interval should be calculated in a single place

Comment (LazyLahtak) [reply packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx]: Done

**Pull request merged**